packages/create-block/index.ts

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env node
+/* eslint-disable no-console */
+const fs = require('fs');
+const prompts = require('prompts');
+const path = require('path');
+const spawn = require('cross-spawn');
+const chalk = require('chalk');
+const commandLineArgs = require('command-line-args');
+const commandLineUsage = require('command-line-usage');
+
+type Options = {
+  name: string;
+  alias?: string;
+  type: StringConstructor | BooleanConstructor;
+  defaultValue?: string;
+  defaultOption?: string;
+  description: string;
+};
+
+type LanguageType = 'typescript' | 'javascript';
+
+/**
+ * Define the command line options.
+ */
+const options: Options[] = [
+  {
+    name: 'namespace',
+    alias: 'n',
+    type: String,
+    defaultValue: 'create-block',
+    defaultOption: 'create-block',
+    description: 'The namespace for the block. (default: create-block)',
+  },
+  {
+    name: 'blocksDir',
+    alias: 'b',
+    description: 'The directory where the blocks will be created relative to the current working directory. (default: blocks)',
+    type: String,
+    defaultValue: 'blocks',
+    defaultOption: 'blocks',
+  },
+  {
+    name: 'blockLanguage',
+    alias: 'l',
+    description: 'The language for the block. Accepts `typescript` or `javascript`',
+    type: String,
+  },
+  {
+    name: 'help',
+    alias: 'h',
+    description: 'Display this usage guide.',
+    type: Boolean,
+  },
+];
+
+/**
+ * Validate the block language.
+ *
+ * @param value - The block language.
+ * @return      - The block language if it is valid.
+ *
+ * @throws {Error} If the block language is not one of the accepted values.
+ */
+function validateBlockLanguage(value: string) {
+  if (value !== 'typescript' && value !== 'javascript') {
+    throw new Error(
+      chalk.red(
+        'The block language must be one of \'typescript\' or \'javascript\'\n',
+      ),
+    );
+  }
+  return value;
+}
+
+// Get the options from the command line.
+const {
+  namespace,
+  blocksDir: blocksDirectory,
+  blockLanguage,
+  help,
+} : {
+  namespace: string;
+  blocksDir: string;
+  blockLanguage: LanguageType | undefined;
+  help: boolean;
+} = commandLineArgs(options);
+
+// Display the help text if the --help option is used.
+const usage = commandLineUsage([
+  {
+    header: 'Alley Create Block',
+    content: 'Alley Create Block is an external template for @wordpress/create-block with set configurations defined for scaffolding a dynamic WordPress block into an existing project that uses WP Scripts (@wordpress/scripts).',
+  },
+  {
+    header: 'Options',
+    optionList: options,
+  },
+]);
+
+// Display the help text if the --help option is used.
+if (help) {
+  console.log(usage);
+  process.exit(1);
```

Comment:
I think this can exit 0 - it's still a successful execution and a non-zero exit status indicates a problem.